### ISSUE TITLE
[Contract Upgrade] e2e test with orders using vault balances

### ISSUE THREAD

**Issue by bh2smith** [gnosis/gp-v2-services#910]
To conclude the integration being performed here, it would be wise to introduce an e2e test which fills orders having vault internal and vault external selected as the balance to and from. One might think this test would be appropriate to include upon the introduction of these fields (#908), but there is other machinery needed before we can do so.

**Comment by nlordell** [comment]
I think this is out of scope for the initial integration where we will only support direct ERC20 allowances to the vault relayer contract and never go through the vault.
